hypertrace-core-graphql-rx-utils/src/main/java/org/hypertrace/core/graphql/rx/NetworkScheduler.java

```diff
@@ -0,0 +1,15 @@
+package org.hypertrace.core.graphql.rx;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.inject.Qualifier;
+
+@Qualifier
+@Target({FIELD, PARAMETER, METHOD})
+@Retention(RUNTIME)
+public @interface NetworkScheduler {}
```

Comment:
I'm not fully convinced about the name actually :) Though these threads are doing network I/O, network in general might mean barebones network communication but here we deal at upper layers.

Reply:
Yeah don't love the name either, open to suggestions.

Reply:
AsyncWorkScheduler? 
Are they used for the resolver work? If so, we can we can call `ResolverWorkScheduler`?

Reply:
We don't have explicit resolvers in graphql-java, the closest equivalent would probably be called a data fetcher:
```
  @GraphQLField
  @GraphQLNonNull
  @GraphQLName(SPANS_QUERY_NAME)
  @GraphQLDataFetcher(SpanFetcher.class) // This guy
  SpanResultSet spans(
      @GraphQLName(TimeRangeArgument.ARGUMENT_NAME) @GraphQLNonNull TimeRangeArgument between,
      @GraphQLName(FilterArgument.ARGUMENT_NAME) List<FilterArgument> filterBy,
      @GraphQLName(OrderArgument.ARGUMENT_NAME) List<OrderArgument> orderBy,
      @GraphQLName(LimitArgument.ARGUMENT_NAME) int limit,
      @GraphQLName(OffsetArgument.ARGUMENT_NAME) int offset);
}
```

The reason I was avoiding async is that pretty much all of the (non-default) schedulers are async, so the difference comes in for their performance characteristics. For things that are cpu bound for example, the computation scheduler takes the underlying core count information into account when allocating threads. For things that are io bound, by default it's unbounded which is what we want here, we just want to put a cap on the max threads. On the other hand, bounds can either be on the work queue or the number of worker threads. Maybe `ThreadBoundedIoScheduler`? Just `BoundedIoSchedulre` ? Would like to resolve this before rolling out the change to the consumers.

Reply:
I think `BoundedIoSchedulre` is better than `NetworkScheduler` so please go ahead with that. Thanks for the details.